File: messageTypes.py
import json
from enum import Enum


class MessageTypes(Enum):
    CREATE_USER = "create_user"


class BaseMessage:
    def __init__(self):
        self.msg = {"type": None, "payload": {}}

    def getMsg(self):
        return self.msg

    def __str__(self):
        return json.dumps(self.msg, indent=1)


class CreateUserMessage(BaseMessage):
    def __init__(self):
        super().__init__()
        self.msg["type"] = MessageTypes.CREATE_USER.value

    def genPayload(self, userName, userID):
        print("ere")
        payload = {"userName": userName, "userID": userID}
        self.msg["payload"] = payload
        return self

File: client/message-types.js
export const CREATE_USER = "create_user";
export const CREATE_ROOM = "create_room";
export const JOIN_ROOM = "join_room";
export const FETCH_ROOMS = "fetch_rooms";
export const START_GAME = "start_game";
export const SUBMIT_ANSWER = "submit_answer";

export const create_user = (userName) => ({
  type: CREATE_USER,
  payload: {
    userName,
  },
});

export const create_room = (userID, roomName) => ({
  type: CREATE_ROOM,
  payload: {
    userID,
    roomName,
  },
});
